File: API_Escola_Stable/Models/Disciplina.cs
using System.ComponentModel.DataAnnotations;
//ALEXANDRE
namespace API_Escola_Stable.Models
{
    public class Disciplina
    {
        [Key]
        public int? IdDisciplina { get; set; }

        public string? NomeDiciplina { get; set; }

        public int? CargaHoraria { get; set; }
        
        public Professor? ProfessorResponsavel { get; set; }

    }

}

File: AppUniversidade/src/app/Disciplina.ts
import { Professor } from "./Professor";

export class Disciplina{
    IdDisciplina: number = 0;
    NomeDiciplina: string = '';
    CargaHoraria: number = 0;
    IdProfessor: number = 0;
    ProfessorResponsavel: Professor | undefined;
}